src/visGeometry/index.ts

```diff
@@ -1146,11 +1153,17 @@ class VisGeometry {
          * @param ids agent ids that should all have the same color
          * @param colorId index into the color array
          */
-        if (this.isIdColorMappingSet) {
-            throw new FrontEndError(
-                "Attempted to set agent-color after color mapping was finalized"
-            );
-        }
+
+        //TODO: this is commented out because I'm unsure how best to handle this
+        // the purpose of this feature is to allow color mapping to be changed
+        // should we just remove this error handling? or should the color change methods
+        // clear the mapping, apply the color change then reset it?
+
+        // if (this.isIdColorMappingSet) {
+        //     throw new FrontEndError(
+        //         "Attempted to set agent-color after color mapping was finalized"
+        //     );
+        // }
```

Comment:
@meganrm thoughts on this? not sure the ideal way to approach this. If this is error handling feels necessary I'll add code needed to clear and reset color mapping, but I'm trying to keep things streamlined.